File: app/src/main/java/com/github/android/lvrn/lvrnproject/persistent/entity/ProfileDependedEntity.java
package com.github.android.lvrn.lvrnproject.persistent.entity;

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 */

public abstract class ProfileDependedEntity extends Entity {

    /**
     * An id of the profile, which the entity is belonged.
     */
    protected String profileId;

    public String getProfileId() {
        return profileId;
    }

    public void setProfileId(String profileId) {
        this.profileId = profileId;
    }

    @Override
    public String toString() {
        return "ProfileDependedEntity{" + super.toString() +
                "profileId='" + profileId + '\'' +
                '}';
    }
}

File: app/src/main/java/com/github/valhallalabs/laverna/persistent/entity/base/ProfileDependedEntity.kt
package com.github.valhallalabs.laverna.persistent.entity.base

/**
 * @author Vadim Boitsov <vadimboitsov1@gmail.com>
 * @author Oleksandr Tyshkovets <olexandr.tyshkovets@gmail.com>
 */
abstract class ProfileDependedEntity : Entity() {
    abstract val profileId: String

    override fun toString(): String {
        return "ProfileDependedEntity{" + super.toString() +
                "profileId=" + profileId +
                "}"
    }

}